contrib/punycode/punycode.h

```diff
@@ -0,0 +1,170 @@
+/*
+punycode-sample.c 2.0.0 (2004-Mar-21-Sun)
+http://www.nicemice.net/idn/
+Adam M. Costello
```

Comment:
I moved Adam's code to the `contrib` folder and added a license there (because it's a third-part code and it's better to separate it from our code).